src/digitalmarketplace/components/cookie-settings/cookie-settings.js

```diff
@@ -0,0 +1,129 @@
+
+  if (!currentConsentCookie) {
+    // Don't populate the form
+    return
+  }
+
+  this.hideWarningMessage()
+
+  // Populate the form with the existing choice
+  var radioButton
+  if (currentConsentCookie.analytics) {
+    radioButton = this.$module.querySelector('input[name=cookies-analytics][value=On]')
+  } else {
+    radioButton = this.$module.querySelector('input[name=cookies-analytics][value=Off]')
+  }
+  radioButton.checked = true
+}
+
+CookieSettings.prototype.submitSettingsForm = function (event) {
+  event.preventDefault()
+
+  var formInputs = event.target.querySelectorAll('input[name=cookies-analytics]')
+  var options = {}
+
+  // Retrieve the selected value from the form inputs
+  for (var i = 0; i < formInputs.length; i++) {
+    var input = formInputs[i]
+    if (input.checked) {
+      var value = input.value === 'On'
+
+      options.analytics = value
+      break
+    }
+  }
+  // the cookie choice must be set when form is submitted
+  if (options.analytics === undefined) {
+    this.showErrorMessage()
+    return false
+  }
+
+  // Set the analytics cookie preferences
+  // If 'Off' option not checked, this function will also delete any existing Google Analytics cookies
+  setConsentCookie(options)
+
+  // If 'On' option checked and analytics not yet present,
+  // initialise Analytics (this includes firing the initial pageview)
+  if (options.analytics && !window.DMGOVUKFrontend.analytics) {
+    InitialiseAnalytics()
+  }
+
+  this.hideWarningMessage()
+  this.hideErrorMessage()
+  this.showConfirmationMessage()
+
+  return false
+}
+
+CookieSettings.prototype.showConfirmationMessage = function () {
+  var confirmationMessage = document.querySelector('#cookie-settings-confirmation')
+  var previousPageLink = document.querySelector('.cookie-settings__prev-page')
+  var referrer = CookieSettings.prototype.getReferrerLink()
+
+  document.body.scrollTop = document.documentElement.scrollTop = 0
+
+  if (referrer && referrer !== document.location.pathname) {
+    previousPageLink.href = referrer
+    previousPageLink.style.display = 'block'
+  } else {
+    previousPageLink.style.display = 'none'
+  }
+
+  confirmationMessage.style.display = 'block'
+}
+
+CookieSettings.prototype.showErrorMessage = function () {
+  var errorMessage = document.querySelector('div#cookie-settings-error')
+  if (errorMessage !== null) {
+    errorMessage.style.display = 'block'
+    document.body.scrollTop = document.documentElement.scrollTop = 0
+  }
+}
+
+CookieSettings.prototype.hideErrorMessage = function () {
+  var errorMessage = document.querySelector('div#cookie-settings-error')
+  if (errorMessage !== null) {
+    errorMessage.style.display = 'none'
+  }
+}
+
+CookieSettings.prototype.hideWarningMessage = function () {
+  var warningMessage = document.querySelector('div#cookie-settings-warning')
+  if (warningMessage !== null) {
+    warningMessage.style.display = 'none'
+  }
+}
```

Comment:
I think we should be making these hidden by adding a class which has `display: none`, instead of setting it directly on the object.

See for instance how this is done in the Design System for the tabs component https://github.com/alphagov/govuk-frontend/blob/master/src/govuk/components/tabs/tabs.js

Reply:
Agreed - is it OK to add that in a separate PR?

Reply:
Sure 👍